### ISSUE TITLE
Unexpected "CS0165: Use of unassigned local variable" for a constructor invocation

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#16348]
**Steps to Reproduce**:
```
public class Cls
{
    public static void Test1()
    {
        Test(out int x1);
    }

    public static void Test2()
    {
        var x = new Cls(out int x2);
    }
}
```

**Actual Behavior**:
```
                // (11,29): error CS0165: Use of unassigned local variable 'x2'
                //         var x = new Cls(out int x2);
                Diagnostic(ErrorCode.ERR_UseDefViolation, "int x2").WithArguments("x2").WithLocation(11, 29)
```

**Expected Behavior**:
No flow analysis errors. Note that a method invocation '''Test(out int x1);``` doesn't trigger similar error.

**Comment by AlekseyTs** [comment]
https://github.com/dotnet/roslyn/pull/16354 adds a test affected by this behavior.

**Comment by jcouv** [comment]
Moved out of RTM because: there is an easy workaround, and we can remove this error later.
When fixing this, please test with discards (`out int _` and `out _`) as well.

**Comment by jcouv** [comment]
The bound tree for the constructor case has a "bad expression" which is broader (covers the entire construction), whereas the tree for method invocation is a proper invocation with arguments and narrower children "bad expressions".

```
//         var x = new Cls(out int x2);

block
├─locals: {Cls x, System.Int32 x2}
├─localFunctions: {}
└─statements
  └─localDeclaration
    ├─localSymbol: Cls x
    ├─declaredType
    │ └─typeExpression
    │   ├─aliasOpt
    │   ├─inferredType: True
    │   ├─boundContainingTypeOpt
    │   └─type: Cls
    ├─initializerOpt
    │ └─badExpression
    │   ├─resultKind: OverloadResolutionFailure
    │   ├─symbols: {Cls..ctor()}
    │   ├─childBoundNodes
    │   │ └─local
    │   │   ├─localSymbol: System.Int32 x2
    │   │   ├─isDeclaration: True
    │   │   ├─constantValueOpt
    │   │   └─type: System.Int32
    │   └─type: Cls
    └─argumentsOpt

//         Test(out int x1);

block
├─locals: {System.Int32 x1}
├─localFunctions: {}
└─statements
  └─expressionStatement
    └─expression
      └─call
        ├─receiverOpt
        │ └─badExpression
        │   ├─resultKind: Empty
        │   ├─symbols: {}
        │   ├─childBoundNodes
        │   └─type: ?
        ├─method: ? ?.()
        ├─arguments
        │ └─local
        │   ├─localSymbol: System.Int32 x1
        │   ├─isDeclaration: True
        │   ├─constantValueOpt
        │   └─type: System.Int32
        ├─argumentNamesOpt: (null)
        ├─argumentRefKindsOpt: {Out}
        ├─isDelegateCall: False
        ├─expanded: False
        ├─invokedAsExtensionMethod: False
        ├─argsToParamsOpt: (null)
        ├─resultKind: NotInvocable
        ├─binderOpt: Microsoft.CodeAnalysis.CSharp.BlockBinder
        └─type: ?
```

**Comment by gafter** [comment]
Note that this is a case of a cascaded diagnostic. The compiler is still correct (it only produces errors on erroneous programs).
